mjolnir/input/read_local_interaction.hpp

```diff
@@ -122,6 +123,136 @@ read_contact_interaction(const std::string& kind, const toml::value& local)
+typename std::unique_ptr<LocalInteractionBase<traitsT>>
+read_contact_in_directional_contact_interaction(const std::string& kind, const toml::value& local)
+{
+  MJOLNIR_GET_DEFAULT_LOGGER();
+  using real_type = typename traitsT::real_type;
+
+  real_type margin = 0.5; // default value
+  if (local.as_table().count("margin") == 1)
+  {
+    margin = toml::find<real_type>(local, "margin");
+  }
+
+  const auto contact_potential = toml::find<std::string>(local, "potentials", "contact");
+  if(contact_potential == "GoContact")
+  {
+    MJOLNIR_LOG_NOTICE("-- contact potential function is 10-12 Go contact.");
+    using contact_potentialT = GoContactPotential<real_type>;
+
+    return make_unique<DirectionalContactInteraction<traitsT, angle1_potentialT,
+               angle2_potentialT, contact_potentialT>>(
+                   kind, read_directional_contact_potentials<
+                   4, real_type, angle1_potentialT,
+                   angle2_potentialT, contact_potentialT>(local));
+  }
+  else if(contact_potential == "Gaussian")
+  {
+    MJOLNIR_LOG_NOTICE("-- contact potential function is Gaussian.");
+    using contact_potentialT = GaussianPotential<real_type>;
+
+    return make_unique<DirectionalContactInteraction<traitsT, angle1_potentialT,
+               angle2_potentialT, contact_potentialT>>(
+                   kind, read_directional_contact_potentials<
+                   4, real_type, angle1_potentialT,
+                   angle2_potentialT, contact_potentialT>(local));
+  }
+  else if(contact_potential == "Uniform")
+  {
+    MJOLNIR_LOG_NOTICE("-- contact potential function is Uniform potential");
+    using contact_potentialT = UniformPotential<real_type>;
+
+    return make_unique<DirectionalContactInteraction<traitsT, angle1_potentialT,
+               angle2_potentialT, contact_potentialT>>(
+                   kind, read_directional_contact_potentials<
+                   4, real_type, angle1_potentialT,
+                   angle2_potentialT, contact_potentialT>(local));
+  }
+  else
+  {
+    throw_exception<std::runtime_error>(toml::format_error("[error] "
+        "mjolnir::read_directional_contact_interaction: invalid contact potential",
+        toml::find<toml::value>(local, "potentials", "contact"), "here", {
+        "expected value is one of the following.",
+        "- \"GoContact\": r^12 - r^10 type native contact potential",
+        "- \"Gaussian\" : well-known gaussian potential"
+        }));
+  }
+}
+
+// This is internal function for read_directional_contact_interaction function.
+template<typename traitsT, typename ... PotentialTs>
+typename std::enable_if<sizeof...(PotentialTs) == 2,
+                        std::unique_ptr<LocalInteractionBase<traitsT>>>::type
+read_angle_in_directional_contact_interaction(
+    const std::string& kind, const toml::value& local, std::vector<std::string>&)
+{
+  return read_contact_in_directional_contact_interaction<traitsT, PotentialTs...>(kind, local);
```

Comment:
You can put branching for contact potential in `read_contact_in_directional_contact_interaction` here, no?